.github/workflows/test.yaml

```diff
@@ -59,7 +59,18 @@ jobs:
       - uses: actions/setup-node@v1
         with:
           node-version: "16.x"
-      - run: cd weave-js && yarn install --frozen-lockfile && yarn eslint && yarn tslint && yarn prettier && yarn run tsc
+      - run: |
+            set -e
+            cd weave-js
+            yarn install --frozen-lockfile
+            yarn eslint
+            yarn tslint
+            yarn prettier-fix
+            git config --local user.email "action@github.com"
+            git config --local user.name "GitHub Action"
+            git diff --quiet && git diff --staged --quiet || (git commit -am "Format code" && git push)
```

Comment:
68-71 actually fix what prettier can fix and commit it back to the branch.

Reply:
Should we be `git push`ing here? Seems like this runs during the commit phase so we should let the user do the pushing?

Reply:
Happy Path:
* user makes a commit
* user has already made necessary changes using prettier
* git diff and --staged in line 71 succeeds, so git commit is not run
* User continues pushing as usual and notices nothing.

Old Path for Target use case:
* user makes a commit which needs prettier changes
* CI notices it and asks user to fix it
* User comes back after some time, fixes prettier and continues

New Path for Target use case:
* user makes a commit which needs prettier changes
* CI notices it and also fixes it
* User needs to `git pull` before pushing
* **If user sees merge conflicts, they can just use `git push -f` and then pull**

Reply:
The New Path feels more complicated to me. I'd prefer to always be in control of what I push to remote. What we do with the python pre-commit is make the mypy/lint fixes, but leave it to the user to commit those changes. 

I think specifically, I'd remove ` && git push` from the PR. This would make the new path:

* user makes a commit which needs prettier changes
* CI notices it and also fixes it
* User git pushes


Notice that the user behavior is no different than the happy path.

Reply:
Tim and I followed up here, and this PR is good to stay as it auto-lints and commits when changes may be made from browser. It would also be good to add a pre-commit hook or something later so that developer machine workflows are also supported, but that's not a blocker.